rib/rib.go

```diff
@@ -237,14 +255,22 @@ type pendingEntry struct {
 	op *spb.AFTOperation
 }
 
-// SetHook assigns the supplied hook to all network instance RIBs within
+// SetPostChangeHook assigns the supplied hook to all network instance RIBs within
 // the RIB structure.
-func (r *RIB) SetHook(fn RIBHookFn) {
+func (r *RIB) SetPostChangeHook(fn RIBHookFn) {
 	for _, nir := range r.niRIB {
+		nir.mu.Lock()
 		nir.postChangeHook = fn
+		nir.mu.Unlock()
 	}
 }
 
+// SetResolvedEntryHook asssigns the supplied hook to all network instance RIBs within
+// the RIB structure.
+func (r *RIB) SetResolvedEntryHook(fn ResolvedEntryFn) {
```

Comment:
nit: Same comment here. Should this be SetIPv4ResolvedEntryHook ?

Reply:
Same comment as above.